plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -895,40 +892,20 @@ private void checkColumnTypes(List<ColumnMetadata> columnMetadata)
         }
     }
 
-    private static boolean deleteRecursivelyIfExists(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path path)
-    {
-        FileSystem fileSystem;
-        try {
-            fileSystem = hdfsEnvironment.getFileSystem(context, path);
-        }
-        catch (IOException e) {
-            LOG.warn(e, "IOException while trying to delete '%s'", path);
-            return false;
-        }
-
-        return deleteIfExists(fileSystem, path, true);
-    }
-
-    private static boolean deleteIfExists(FileSystem fileSystem, Path path, boolean recursive)
+    private static void deleteRecursivelyIfExists(TrinoFileSystem fileSystem, Path path)
```

Comment:
This method can be replaced with a call to
```
fileSystem.deleteDirectory()
```
It can be called when the directory doesn't exist. (we should write Javadoc for the expected behavior of `TrinoFileSystem` ... this is my fault)

Reply:
Sorry, I should have read the code carefully. Remove the directory check, but left this method because it's called form lambda.